main.tf

```
@@ -78,12 +79,17 @@ resource "google_compute_health_check" "tcp" {
     port_name    = var.health_check["port_name"]
     proxy_header = var.health_check["proxy_header"]
   }
+
+  log_config {
```

We should make this a dynamic block.

```suggestion
  dynamic "log_config" {
    for_each = var.health_check["enable_log"] ? [true] : []
    content {
      enable = true
    }
  }
```